zerver/models.py

```diff
@@ -4659,13 +4672,20 @@ class RealmAuditLog(AbstractRealmAuditLog):
         null=True,
         on_delete=CASCADE,
     )
+    modified_user_group = models.ForeignKey(
+        UserGroup,
+        null=True,
+        on_delete=CASCADE,
+    )
```

Comment:
Can you start a #backend thread about whether these should be `CASCADE`? I'm not entirely sure, and changing that would be a separate PR, but it seems like a question worth asking and then documenting the answer in a comment in this definition.

Reply:
Posted a discussion [here](https://chat.zulip.org/#narrow/stream/3-backend/topic/Avoiding.20ON.20DELETE.20CASCADE.20on.20auditlogs).